allium/lib/operator_analysis.py

```
@@ -23,46 +23,151 @@
+            bucket = index.setdefault(leader_contact, [])
+            bucket.append(ranking)
+
+    # Each contact's bucket is sorted by category-iteration order; the
+    # legacy implementation sorted by rank ascending. Apply the same
+    # final sort to preserve identical output for downstream templates.
+    for rankings in index.values():
+        rankings.sort(key=lambda x: x['rank'])
+    return index
+
+
+def _get_contact_rankings_index(relay_set):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_get_contact_rankings_index`

(ANN202)